packages/mask/src/components/shared/ApplicationSettingPluginList.tsx

```diff
@@ -0,0 +1,161 @@
+    entry: Plugin.SNSAdaptor.ApplicationEntry
+    pluginId: string
+    enabled?: boolean
+}
+
+// #region kv storage
+export function setUnlistedApp(app: Application, unlisted: boolean) {
+    PersistentStorages.ApplicationEntryUnListedList.storage[app.entry.ID].setValue(unlisted)
+}
+
+export function getUnlistedApp(app: Application): boolean {
+    return PersistentStorages.ApplicationEntryUnListedList.storage[app.entry.ID].value
+}
+// #endregion
+
+const useStyles = makeStyles()((theme) => ({
+    list: {
+        display: 'grid',
+        gap: theme.spacing(2, 1),
+        gridTemplateColumns: 'repeat(6, 1fr)',
+        gridTemplateRows: 'repeat(3, 1fr)',
+        width: '100%',
+        padding: '8px 16px 0px 0px',
+        overflowY: 'scroll',
+        height: 238,
+    },
+    listItem: {
+        width: 84,
+        height: 56,
+        padding: 0,
+        background: theme.palette.background.default,
+        justifyContent: 'center',
+        alignItems: 'center',
+        cursor: 'pointer',
+        borderRadius: 8,
+        position: 'relative',
+    },
+    iconWrapper: {
+        '> *': {
+            width: 36,
+            height: 36,
+        },
+    },
+    loadingWrapper: {
+        display: 'flex',
+        height: 400,
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    unlisted: {
+        fontSize: 18,
+        fontWeight: 600,
+    },
+    placeholderWrapper: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: 238,
+    },
+    placeholder: {
+        color: getMaskColor(theme).textLight,
+    },
+}))
+
+export function ApplicationSettingPluginList() {
+    const { classes } = useStyles()
+    const { t } = useI18N()
+    const snsAdaptorPlugins = useActivatedPluginsSNSAdaptor('any')
+    const applicationList = useMemo(
+        () =>
+            snsAdaptorPlugins
+                .reduce<Application[]>((acc, cur) => {
+                    if (!cur.ApplicationEntries) return acc
+                    return acc.concat(
+                        cur.ApplicationEntries.filter((x) => x.appBoardSortingDefaultPriority).map((x) => {
+                            return {
+                                entry: x,
+                                pluginId: cur.ID,
+                            }
+                        }) ?? [],
+                    )
+                }, [])
+                .sort(
+                    (a, b) =>
+                        (a.entry.appBoardSortingDefaultPriority ?? 0) - (b.entry.appBoardSortingDefaultPriority ?? 0),
+                ),
+        [snsAdaptorPlugins],
+    )
+    const [listedAppList, setListedAppList] = useState(applicationList.filter((x) => !getUnlistedApp(x)))
+    const [unlistedAppList, setUnListedAppList] = useState(applicationList.filter((x) => getUnlistedApp(x)))
+
+    useEffect(() => {
+        return MaskMessages.events.__kv_backend_persistent__.on((data) => {
```

Comment:
You should not listen to this event.

Reply:
Why, then how to notice the component to re-render?

Reply:
This is an internal event, and should not be used by other normal components. Define `setUnlistedApp` in this function component, and when it is called, update the state.

Reply:
done